src/FESpaces/DiscreteModelWithFEMaps.jl

```diff
@@ -0,0 +1,156 @@
+"""
+Given a Discrete Model and a reffe, builds a new grid in which the geometrical
+is a `FEFunction`. This is useful when considering geometrical maps that are the
+result of a FE problem (mesh displacement).
+"""
+struct GridWithFEMap{Dc,Dp,A,B,C,D,E,F} <: Grid{Dc,Dp}
+  grid::Grid{Dc,Dp}
+  fe_sp::A
+  scal_fe_sp::B
+  fe_map::C
+  node_coords::D
+  reffes::E
+  cell_type::F
+end
+
+function GridWithFEMap(model,reffe::ReferenceFE; kwargs...)
```

Comment:
I have a concern here. `model` is such that there is a (potential) polytope associated to each of its cells. E.g., one may have a mixed model, with TETs and QUADs.

On the other hand, `reffe` also is built atop polytope. Should we check that the polytopes of associated to the cells of the model and the one of `reffe` match?